pkgs/development/python-modules/lm-format-enforcer/default.nix

```diff
@@ -0,0 +1,42 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pydantic,
+  interegular,
+  pyyaml,
+  poetry-core,
+}:
+
+buildPythonPackage rec {
+  pname = "lm-format-enforcer";
+  version = "0.10.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "noamgat";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GOnMj910rgzYeIeN2yLcXZDDel/Hu6nv7ov5BrlHJLg=";
+  };
+
+  build-system = [ poetry-core ];
+
+  propagatedBuildInputs = [
+    interegular
+    pydantic
+    pyyaml
+  ];
+
+  doCheck = false; # most tests require internet access
+
+  pythonImportChecks = [ "lmformatenforcer" ];
```

Comment:
*Check singular; verify `nix log` for `lmformatenforcer` actually being imported

Reply:
Imports plural:)

Reply:
in general, grep nixpkgs for examples

Reply:
oops, fixing.

Reply:
I blame nix for this one, because the inputs didn't change this didn't rebuild when I test it locally.